File: 202_HappyNumber/c++/HappyNumber.cc
class Solution {
public:
    bool isHappy(int n) {
        set<int> seen;
        string numStr;
        while(seen.count(n)==0){
            seen.insert(n);
            numStr = to_string(n);
            n = 0;
            for(auto i:numStr){
                n+=int(i-'0')*int(i-'0');
            }
        }
        
        return n==1;
    }
};

File: 202_HappyNumber/c#/HappyNumber.cs
public class Solution {
    public bool IsHappy(int n) {
        HashSet<int> seen = new HashSet<int>();
        String numStr;
        while(!seen.Contains(n)){
            seen.Add(n);
            numStr = n.ToString();
            n = 0;
            foreach(var i in numStr){
                n+=(i-'0')*(i-'0');
            }
        }
        
        return n==1;
    }
}